onnx_caffe2/backend.py

```diff
@@ -297,7 +300,71 @@ def _create_gemm(cls, n, opset_version):
         return ops
 
     @classmethod
-    def _create_pad(cls, n, opset_version):
+    def _create_lstm(cls, onnx_model, n, opset_version):
+        assert onnx_model is not None, "cannot convert LSTMs through this codepath"
+
+        attrs = dict(n.attrs)
+        hidden_size = attrs.pop('hidden_size')
+        assert not attrs, "unsupported LSTM attributes: " + attrs.keys()
+
+        input_blob, W, R, B, sequence_lens, initial_h, initial_c = n.inputs
+
+        # ad-hoc, informally-specified, bug-ridden, slow
+        # implementation of shape inference
+        for x in onnx_model.graph.input:
```

Comment:
so what about analyzing weight shapes instead? the current thing is super sketchy and cover fewer cases as the input to LSTM layer isn't necessarily the graph input or another LSTM

Reply:
yeah I tried briefly and it wasn't quite trivial (I think because the init-net-splitting means that the weights are not accessible in the same onnx model as the LSTM op is), but I'll try again.